galaxy_api/api/v3/viewsets.py

```diff
@@ -124,9 +125,17 @@ def retrieve(self, request, *args, **kwargs):
             name=self.kwargs['name'],
             version=self.kwargs['version'],
         )
-        response['download_url'] = request.build_absolute_uri(response['download_url'])
+        response['download_url'] = self._transform_pulp_url(request, response['download_url'])
```

Comment:
@cutwater Before the change the `download_url` in response looked like this to me:

`http://automation-hub:24816/api/automation-hub/v3/artifacts/collections/rblanco.collection_demo.1.0.17`

I've been getting `Not Found` error.

After the change, the `download_url` in response looks like this:

`http://localhost:5001/api/automation-hub/v3/artifacts/collections/rblanco.collection_demo.1.0.17`.

Trying to access it, I'm getting 403 error, even though I'm logged in as user in partner engineer group.

`{"errors":[{"status":"403","code":"not_authenticated","title":"Authentication credentials were not provided."}]}`.

What is the expected behaviour here?

Reply:
Can you please describe your development environment you use for testing?

Reply:
Accessing API of Automation Hub behind proxy: http://ci.foo.redhat.com:5001/api/automation-hub/v3/artifacts/collections/rblanco.collection_demo.1.0.17

notes:

```diff
-const localhost = (process.env.PLATFORM === 'linux') ? 'localhost' : 'host.docker.internal';
+const localhost = (process.env.PLATFORM === 'linux') ? 'ci.foo.redhat.com' : 'host.docker.internal';
```